aspnetcore/performance/caching/memory.md

```diff
@@ -1,164 +1,111 @@
+The following code uses [TryGetValue](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.imemorycache) to cache and fetch the current time:
 
-[!code-csharp[Main](memory/sample/test/CachingSample.Tests/MemoryCacheTests.cs?highlight=6-11,18&range=22-41)]
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet1)]
 
-The callback is run on a different thread from the code that removes the item from the cache.
+The current time and the cached time is displayed:
 
->[!WARNING]
-> If the callback is used to repopulate the cache it is possible other requests for the cache will take place (and find it empty) before the callback completes, possibly resulting in several threads repopulating the cached value.
+[!code-html[Main](memory/sample/webcache/views/home/index.cshtml)]
 
-Possible `eviction reasons` are:
+The `DateTime.Now` value will stay in the cache while there are requests within the timeout period (and not enough memory pressure to evict the cached entry). The image below shows the cached value is older than the current value:
 
-**None**
+![Index view with two different times displayed](memory/_static/time.png)
 
-No reason known.
+The following code uses [GetOrCreate](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) and [GetOrCreateAsync](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) to cache data. 
 
-**Removed**
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet2&highlight=3,13-14)]
 
-The item was manually removed by a call to `Remove()`
+The following code calls [Get](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) to fetch the cached time:
 
-**Replaced**
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet_gct)]
 
-The item was overwritten.
+See [IMemoryCache methods](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.imemorycache) and [CacheExtensions methods](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) for a description of the cache methods.
 
-**Expired**
+## Using `MemoryCacheEntryOptions`
 
-The item timed out.
+The following sample:
 
-**TokenExpired**
+- Sets the absolute expiration time. This is the maximum time the entry can be cached and prevents the item from becoming too stale when the sliding expiration is continuously renewed.
+- Sets a sliding expiration time. Requests that access this cached item will reset the sliding expiration clock.
+- Sets the [cache priority](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheitempriority) to `CacheItemPriority.NeverRemove`. 
```

Comment:
Already linked to cache priority previously in this doc, so is this still needed?